base/float.jl

```diff
@@ -532,16 +538,17 @@ significand_mask(::Type{Float32}) = 0x007f_ffff
 
 ## Array operations on floating point numbers ##
 
-float{T<:AbstractFloat}(A::AbstractArray{T}) = A
-
-function float{T}(A::AbstractArray{T})
-    if !isleaftype(T)
-        error("`float` not defined on abstractly-typed arrays; please convert to a more specific type")
-    end
-    convert(AbstractArray{typeof(float(zero(T)))}, A)
+# float, broadcast over arrays
+broadcast{T<:AbstractFloat}(::typeof(float), A::AbstractArray{T}) = A
```

Comment:
Probably we should also have `broadcast{T<:AbstractFloat}(::typeof(float), A::AbstractArray{Complex{T}}) = A` in complex.jl